src/bm_sim/checksums.cpp

```diff
@@ -149,7 +149,12 @@ CalcBasedChecksum::update_(Packet *pkt) const {
 bool
 CalcBasedChecksum::verify_(const Packet &pkt) const {
   const uint64_t cksum = calculation->output(pkt);
+  std::stringstream ss, ess;
+  ss << std::hex << cksum;
   const auto &f_cksum = pkt.get_phv()->get_field(header_id, field_offset);
+  ess << std::hex << f_cksum.get<uint64_t>();
+  BMLOG_DEBUG_PKT(pkt, "Checksum '{}': computed {} : in pkt {}",
```

Comment:
please do the conversion from `uint64` to hex string in a separate helper function. And make sure the function is only called when invoking the `BMLOG_DEBUG_PKT`. I want to make sure that the function is not evaluated when the debugging macros are turned off, even though the compiler should take care of that in O2.